src/common/types/validity_mask.cpp

```diff
@@ -46,6 +46,20 @@ string ValidityMask::ToString(idx_t count) const {
 	return result;
 }
 
+bool ValidityMask::Equal(ValidityMask &mask, idx_t count) const {
+	if (AllValid() && mask.AllValid()) {
+		return true;
+	} else if (!AllValid() && !mask.AllValid()) {
+		auto v_count = EntryCount(count);
+		for (idx_t entry_idx = 0; entry_idx < v_count; entry_idx++) {
+			mask.validity_mask[entry_idx] = validity_mask[entry_idx];
+		}
+		return true;
+	} else {
+		return false;
```

Comment:
`AllValid` is a (poorly named?) alias for `HasEntry`, all entries can still be valid even if `AllValid` returns false, so a check still needs to be performed here.